### PR TITLE
Fixes conan-io#4674.  Adds --sysconfdir=/etc to configuration settings.

### PR DESCRIPTION
**Pull request by sourcedelica**
Specify library name and version:  **odbc/2.3.7**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by uilianries** [comment]
Why don't you use self.package_folder/res instead?

**Comment by sourcedelica** [comment]
Not sure what you mean?

**Review by uilianries — APPROVED** [review]

**Comment by uilianries** [comment]
Forget about. Pointing to conan cache would create a new problem. Better keeping it as system folder.

**Comment by sourcedelica** [comment]
Cool - yes - these config files will be modified by users using binaries where the library was statically linked and there is no Conan cache anymore.  And the recipe doesn't package any config files.

File: recipes/odbc/all/conanfile.py [hunk 1]
```diff
@@ -57,7 +57,8 @@ def _configure_autotools(self):
         args = ["--enable-static=%s" % static_flag,
                 "--enable-shared=%s" % shared_flag,
                 "--enable-ltdl-install",
-                "--enable-iconv=%s" % libiconv_flag]
+                "--enable-iconv=%s" % libiconv_flag,
+                "--sysconfdir=/etc"]
```
Comment (mathbunnyru) [inline recipes/odbc/all/conanfile.py:61]: Is it always `/etc/` on all Linux-like or BSD-like machines?

Comment (sourcedelica) [reply recipes/odbc/all/conanfile.py:61]: That is the default

Comment (sourcedelica) [reply recipes/odbc/all/conanfile.py:61]: When you install it as a system package it includes shared libs and the config file location is `/etc`.  This is making the Conan-provided static libs compatible.  The other alternative is to create an option for this and make the default `/etc`.

**Comment by prince-chrismc** [comment]
Can you please add "Fixes #4674" the the opening comment?

**Review by prince-chrismc — APPROVED** [review]
I have no objection but I am really curious what is it using this folder for?  What config files is it loading at run time?

**Comment by sourcedelica** [comment]
Hi @prince-chrismc -

Changed the comment.

Here is info on the config files:
Where are ODBC drivers defined?  https://www.easysoft.com/developer/interfaces/odbc/linux.html#driver_defn_unixodbc
Where are ODBC data sources defined?  https://www.easysoft.com/developer/interfaces/odbc/linux.html#dsn_defn_unixodbc

**Comment by sourcedelica** [comment]
On it.

**Comment by sourcedelica** [comment]
I'm not sure what's going on.  When I build with hooks on MacOS it works fine.  When I build one of the variants that failed testing (for example, gcc10 using `conanio/gcc10` and the profile from CI) without hooks it builds and tests successfully.  For some reason with hooks enabled the package is created but there are no libraries listed in the test_package's `conanbuildinfo.{cmake,txt}`.  

When the test runs these are displayed:
```
[HOOK - conan-center.py] post_package_info(): [CMAKE FILE NOT IN BUILD FOLDERS (KB-H019)] OK
[HOOK - conan-center.py] post_package_info(): [LIBRARY DOES NOT EXIST (KB-H054)] OK
odbc/2.3.7: Appending PATH environment variable: /home/conan/.conan/data/odbc/2.3.7/_/_/package/a6346e2a1eebddc217148223d48750ebad5e746b/bin
[HOOK - conan-center.py] post_package_info(): [CMAKE FILE NOT IN BUILD FOLDERS (KB-H019)] OK
[HOOK - conan-center.py] post_package_info(): WARN: [LIBRARY DOES NOT EXIST (KB-H054)] Component odbc::odbc libdir "lib" does not contain any declared library
[HOOK - conan-center.py] post_package_info(): [LIBRARY DOES NOT EXIST (KB-H054)] OK
```
I'm assuming it's related to this one in particular:
```
[HOOK - conan-center.py] post_package_info(): WARN: [LIBRARY DOES NOT EXIST (KB-H054)] Component odbc::odbc libdir "lib" does not contain any declared library
```
But the libraries do exist:
```
conan@aa9af32d4e38:~/conan-center-index/recipes/odbc/all$ ls /home/conan/.conan/data/odbc/2.3.7/_/_/package/a6346e2a1eebddc217148223d48750ebad5e746b/lib
libltdl.a  libodbc.a  libodbccr.a  libodbcinst.a
```
I noticed that warning `KB-H054` is not in the docs, maybe it's new and has a bug?

**Comment by mathbunnyru** [comment]
There was a hotfix 2 hours ago, could you update hooks locally and try it out?
https://github.com/conan-io/hooks/commit/c534ca20ed91f25327273d9a7a17aae70e13f477

**Comment by sourcedelica** [comment]
Yep - that fixed it.

**Comment by mathbunnyru** [comment]
You need to close the PR, wait for at least 10 seconds and open it again - this will retrigger the CI.

File: recipes/odbc/all/conanfile.py (hunk 1 above)
Comment (mathbunnyru) [reply recipes/odbc/all/conanfile.py:61]: I think having `/etc` is okay.
If someone really wants to change the value, they will create PR allowing any path.

**Review by mathbunnyru — APPROVED** [review]

File: recipes/odbc/all/conanfile.py (hunk 1 above)
Comment (uilianries) [reply recipes/odbc/all/conanfile.py:61]: I would use self.package_folder/res, but it's complicated to move configuration files from a system to Conan, so /etc is fine for me.

**Review by uilianries — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Comment by sourcedelica** [comment]
Thanks guys!

**Pull request merged**